### PR TITLE
Fix psalm issue about userCounts

### PR DESCRIPTION
**Pull request by CarlSchwan**
Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Review by szaimen — APPROVED** [review]
seems to do the trick 👍

**Review by szaimen — COMMENTED** [review]
I guess we should add this to breaking changes for NC24?

File: apps/user_ldap/lib/User_LDAP.php [hunk 1]
```diff
@@ -47,9 +47,10 @@
 use OCP\IConfig;
 use OCP\IUserSession;
 use OCP\Notification\IManager as INotificationManager;
+use OCP\User\Backend\ICountUsersBackend;
 use Psr\Log\LoggerInterface;
 
-class User_LDAP extends BackendUtility implements \OCP\IUserBackend, \OCP\UserInterface, IUserLDAP {
+class User_LDAP extends BackendUtility implements \OCP\IUserBackend, \OCP\UserInterface, IUserLDAP, ICountUsersBackend {
```
Comment (come-nc) [inline apps/user_ldap/lib/User_LDAP.php:53]: Add use for \OCP\UserInterface and \OCP\IUserBackend as well maybe?

**Review by come-nc — APPROVED** [review]
Can you detail how `implementsActions()` and the `I*Backend` interacts?
Is one the old way of doing thing and one the new way?

Why is `implementsActions(Backend::COUNT_USERS)` used and not `instanceof ICountUsersBackend`

**Comment by CarlSchwan** [comment]
> Can you detail how `implementsActions()` and the `I*Backend` interacts? Is one the old way of doing thing and one the new way?
> 
> Why is `implementsActions(Backend::COUNT_USERS)` used and not `instanceof ICountUsersBackend`

OCP\User\Backend seems to be the replacement of OC_User_Backend (a.k.a very old) and ICountUsersBackend seems to be the modern alternative introduced by 226439a31c836a327ebd8b6a3171bcffa7250597 ihmo `OCP\User\Backend` should probably be deprecated

**Comment by CarlSchwan** [comment]
s3 ci failures are unrelated

**Pull request merged**